server/datasources.development.js

```diff
@@ -1,18 +1,26 @@
-module.exports = {
-    mail: {
-        connector: 'mail',
-        transport: {
-            type: 'smtp',
-            host: 'localhost',
-            secure: false,
-            port: 1025,
-            tls: {
-                rejectUnauthorized: false
-            }
-        },
-        auth: {
-            user: 'test',
-            pass: 'test'
-        }
+const debug = require('debug')('fcc:server:datasources');
+const dsLocal = require('./datasources.local');
+
+// use [MailHog](https://github.com/mailhog/MailHog)
+let mail = {
+  connector: 'mail',
+  Transport: {
+    type: 'smtp',
+    host: 'localhost',
+    secure: false,
+    port: 1025,
+    tls: {
+      rejectUnauthorized: false
     }
+  },
+  auth: {
+    user: 'test',
+    pass: 'test'
+  }
 };
+// use AWS SES if defined
+if (process.env.SES_ID) {
+  debug('using AWS SES to deliver emails');
+  mail = dsLocal.mail;
+}
+module.exports = { mail };
```

Comment:
This will unfortunately not work, crashing the server if SES keys are not available, in development.

> https://loopback.io/doc/en/lb3/Environment-specific-configuration.html#data-source-configuration
> You can override values set in datasources.json in the following files:
> - datasources.local.js or datasources.local.json
> - datasources.env.js or datasources.env.json, where env is the value of NODE_ENV environment variable (typically development or production). For example, datasources.production.json.

This means that in our case `NODE_ENV` needs to be set explicitly to either one `local` or `development` or `production`.

By default when not set, loopback merges everything. Its kind of weird to understand. 

But consider this instead:
```js
const debug = require('debug')('fcc:server:datasources');
const secrets = require('../config/secrets');

// use AWS SES if defined
const sesConfig = {
  connector: 'mail',
  transport: {
    type: 'ses',
    accessKeyId: process.env.SES_ID,
    secretAccessKey: process.env.SES_SECRET
  }
}
// use [MailHog](https://github.com/mailhog/MailHog)
const mailhogConfig = {
  connector: 'mail',
  transport: {
      type: 'smtp',
      host: 'localhost',
      secure: false,
      port: 1025,
      tls: {
          rejectUnauthorized: false
      }
  },
  auth: {
      user: 'test',
      pass: 'test'
  }
};

let datasourcesConfig = {
  db: {
    connector: 'mongodb',
    connectionTimeout: 10000,
    url: secrets.db
  }
};

if(process.env.SES_ID){
  debug('using AWS SES to deliver emails');
  datasourcesConfig.mail = sesConfig;
} else {
  debug('using MailHog to catch emails');
  datasourcesConfig.mail = mailhogConfig;
}

module.exports = datasourcesConfig;
```

And with that we are able to remove the `datasources.local.js` completely and avoid all confusion.

Reply:
Update: 

I had mentioned the file names incorrectly. 
We want everything in the `datasources.local.js` and delete the `datasources.development.js`

Reply:
@raisedadead Thanks for finding this issue. Yes you are correct, although I assumed that the dev settings would overwrite the local settings this is not the case. I've opted to use a dev and a production file to prevent production from ever seeing the mailhog settings.

I've tested both mail options and I have tested production mode. Everything seems to working now. 

For new users they have the option of using MailHog or getting SES keys set up. We can also use the dev file to look for google gmail keys as well if we wanted to give users the option. For now I wanted to get SES keys working as this is what @QuincyLarson and I use during testing.

Thank again for the QA!